### PR TITLE
enable pod gc in gce with threshold of 12500

### LINKED ISSUE #7660: Reap/GC terminated pods
So that successful/failed pods don't accumulate forever.

### PR DESCRIPTION
**Pull request by mikedanese**
Fixes #7660

@alex-mohr @bgrant0607

### CODE REVIEW

**Comment by mikedanese** [comment]
@k8s-bot ok to test

File: cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest [hunk 1]
```diff
@@ -19,6 +20,9 @@
 {% if pillar['enable_deployments'] is defined -%}
   {% set enable_deployments = "--enable-deployment-controller=" + pillar['enable_deployments'] -%}
 {% endif -%}
+{% if grains.cloud in [ 'gce' ] and grains.cloud_config is defined -%}
```
Comment (alex-mohr) [inline cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest]: apologies if this is obvious to people who grok salt -- I don't see why cloud_config needs to be defined in addition to the cloud==gce test?

But then, I'm not sure why this is behavior we'd only want for GCE -- presumably AWS and GKE and bare metal clusters should also not fall over as the number of terminated pods goes to infinity?  What's the justification for GCE-only?

Comment (mikedanese) [inline cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest]: I presumed we only wanted this for gce/gke from reading discussion #7660. Alternatively we can just change the default value of the flag (and not touch salt) then add this to the release notes.

Comment (alex-mohr) [inline cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest]: Changing default SGTM -- fewer special cases where k8s behaves in way X in environment Y but in a different way in Z generally good?  I skimmed #7660 and didn't see a push for GCE-only, but maybe missed something.  Pull in anyone who'd potentially object strongly?

Comment (mikedanese) [inline cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest]: @kubernetes/rh-cluster-infra @smarterclayton @kubernetes/goog-control-plane any objections to make this the flag's default?

Comment (bgrant0607) [inline cluster/saltbase/salt/kube-controller-manager/kube-controller-manager.manifest]: cc @jbeda

**Comment by smarterclayton** [comment]
No objection from me

**Comment by davidopp** [comment]
Should you wait to enable until we have an e2e test?

**Comment by bgrant0607** [comment]
We could just set it to a very small value in our e2e tests.

**Comment by mikedanese** [comment]
What do we want to test? That it does what we expect it to do? Or that the system can handle a high rate of terminated pod creation with a threshold of 12500?

**Comment by davidopp** [comment]
I was thinking set TerminatedPodGCThreshold to some very low value in a test where you create a bunch of pods that get killed immediately by kubelet, and then verify that you end up with TerminatedPodGCThreshold number of pods, not more.

**Comment by mikedanese** [comment]
@davidopp see #15428

**Comment by davidopp** [comment]
LGTM

**Pull request merged**